### PR TITLE
Search SDK: Adding all tests from private repo

### PR DESCRIPTION
**Pull request by bitsforthought**
This was mostly straightforward, except I had to override `WithHandler` to get the tests to pass. Otherwise, `AddMockHandler` would call `ServiceClient.WithHandler`, which calls `Activator.CreateInstance`, which fails because `SearchIndexClient` and `SearchServiceClient` don't have public default constructors.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@brjohnstmsft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (brjohnst). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/Search/Search.Management.Tests/packages.config [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
```
Comment (yugangw-msft) [inline src/Search/Search.Management.Tests/packages.config]: (The git web page can't show all of the changes here, so my suggestion might not help)
if you depend on the TestDependencies like most other ones do, then you don't need to have this file.

File: src/Search/Search.Management.Tests/app.config [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
```
Comment (yugangw-msft) [inline src/Search/Search.Management.Tests/app.config]: you can get rid of this file as well, unless there is common nuget scenario which causes this file to get added back automatically, which i understand :-)

**Comment by yugangw-msft** [comment]
could you also address the CI failure which comes from the generated code you ported over

**Comment by yugangw-msft** [comment]
Search.Management.Tests.csproj should not build for portable, rather only build for net45, unless you do want to. If you uncheck it in the solution configuration manager, that should fix the CI failure

**Pull request merged**